components/user/UserProfile.js

```diff
@@ -87,6 +88,24 @@ function UserProfile({ BASE_URL, data }) {
         </div>
       )}
 
+      {data.isStatsPublic && (
+        <BasicCards
+          data={[
+            {
+              name: "Rank",
+              current: data.profilStats?.profile?.rank,
```

Comment:
Now the typo was fixed, you also need to fix it here also 

```suggestion
              current: data.profileStats?.profile?.rank,
```